plugin.video.disclosetv/resources/lib/disclosetv.py

```diff
@@ -0,0 +1,189 @@
+        soup   = BeautifulSoup(self.openURL(url), "html.parser")
+        videos = soup('div', {'class': 'teaser teaser--third'})
+        for video in videos:
+            try: thumb = 'http:%s'%(video('div', {'class': 'ratio-container ratio16_9'})[0].find('img').attrs['data-src'])
+            except: thumb = FANART
+            items   = video('div', {'class': 'teaser__caption'})
+            vid_url = BASE_URL + (items[0]('a', {'class': 'article-link'})[0].attrs['href'])
+            label   = items[0]('a', {'class': 'article-link'})[0].get_text()
+            timeago = items[0]('span', {'class': 'meta-timeago'})[0].get_text()
+            plot    = '%s - %s'%(timeago, label)
+            try: genre = video('span', {'class': 'teaser-figure__cat'})[0].get_text()
+            except: genre = 'Unknown'
+            runtime = (video('span', {'class': 'teaser-figure__len'})[0].get_text()).split(':')
+            if len(runtime) == 3:
+                h, m, s = runtime
+                duration  = int(h) * 3600 + int(m) * 60 + int(s)
+            else:
+                m, s = runtime
+                duration  = (int(m) * 60) + int(s)
+            infoLabels = {"mediatype":"episode","label":label ,"title":label,"duration":duration,"plot":plot}
+            infoArt    = {"thumb":thumb,"poster":thumb,"fanart":FANART,"icon":ICON,"logo":ICON}
+            self.addLink(label, vid_url, 9, infoLabels, infoArt, len(videos))
+        next = soup('li', {'class': 'more-container__button m-auto'})
+        if len(next) == 0: return
+        next_url   = BASE_URL + next[0].find('a').attrs['href']
+        next_label = (next[0].find('a').attrs['title'] or next[0].get_text())
+        self.addDir(next_label, next_url, 1)
+            
+            
+    def playVideo(self, name, url, liz=None):
+        log('playVideo')
+        info = getVideoInfo(url,QUALITY,True)
+        if info is None: return
+        info = info.streams()
+        url  = info[0]['xbmc_url']
+        liz  = xbmcgui.ListItem(name, path=url)
+        if 'subtitles' in info[0]['ytdl_format']: liz.setSubtitles([x['url'] for x in info[0]['ytdl_format']['subtitles'].get('en','') if 'url' in x])
+        xbmcplugin.setResolvedUrl(int(sys.argv[1]), True, liz)
+        
+        
+    def addYoutube(self, name, url):
+        liz=xbmcgui.ListItem(name)
+        liz.setProperty('IsPlayable', 'false')
+        liz.setInfo(type="Video", infoLabels={"label":name,"title":name} )
+        liz.setArt({'thumb':ICON,'fanart':FANART})
+        xbmcplugin.addDirectoryItem(handle=int(sys.argv[1]),url=url,listitem=liz,isFolder=True)
+        
+           
+    def addLink(self, name, u, mode, infoList=False, infoArt=False, total=0):
+        name = name.encode("utf-8")
+        log('addLink, name = ' + name)
+        liz=xbmcgui.ListItem(name)
+        liz.setProperty('IsPlayable', 'true')
+        if infoList == False: liz.setInfo(type="Video", infoLabels={"mediatype":"video","label":name,"title":name})
```

Comment:
you can probably add the mimetype property (if you know the video format) in the future to avoid kodi to issue head requests and the speed things up.
Cheers and thanks for the submission